backend/apid/middlewares/authorization_attributes.go

```diff
@@ -56,129 +55,32 @@ func (a AuthorizationAttributes) Then(next http.Handler) http.Handler {
 		attrs.APIVersion = vars["version"]
 		attrs.Namespace = vars["namespace"]
 		attrs.Resource = vars["resource"]
-		attrs.ResourceName = vars["name"]
-
-		if attrs.Verb == "get" && attrs.ResourceName == "" {
-			attrs.Verb = "list"
-		}
-
-		// Add the user to the attributes
-		if err := getUser(ctx, attrs); err != nil {
-			http.Error(w, err.Error(), http.StatusUnauthorized)
-			return
-		}
-	})
-}
-
-// LegacyAuthorizationAttributes is an HTTP middleware that populates a context
-// for the request, to be used by other middlewares. You probably want this
-// middleware to be executed early in the middleware stack.
-//
-// This middleware is here purely to support, old, pre-API versioning routes. It
-// should be removed when all the old routes are replaced by versioned ones.
-//
-// The expected path is one of the following:
-// /assets
-// /assets/{id}
-// /checks
-// /checks/{id}
-// /checks/{id}/execute
-// /checks/{id}/hooks/{type}
-// /checks/{id}/hooks/{type}/hook/{hook}
-// /cluster/members
-// /cluster/members/{id}
-// /entities
-// /entities/{id}
-// /filters
-// /filters/{id}
-// /events
-// /events/{entity}
-// /events/{entity}/{check}
-// /extensions
-// /extensions/{id}
-// /handlers
-// /handlers/{id}
-// /hooks
-// /hooks/{id}
-// /mutators
-// /mutators/{id}
-// /rbac/namespaces
-// /rbac/namespaces/{id}
-// /rbac/users
-// /rbac/users/{id}
-// /silenced
-// /silenced{id}
-// /silenced/subscriptions
-// /silenced/subscriptions/{subscription}
-// /silenced/checks
-// /silenced/checks/{check}
-type LegacyAuthorizationAttributes struct{}
-
-func (a LegacyAuthorizationAttributes) Then(next http.Handler) http.Handler {
-	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		ctx := r.Context()
-		attrs := &authorization.Attributes{}
-
-		ctx = authorization.SetAttributes(ctx, attrs)
-		defer next.ServeHTTP(w, r.WithContext(ctx))
-
-		switch r.Method {
-		case "POST":
-			attrs.Verb = "create"
-		case "GET", "HEAD":
-			attrs.Verb = "get"
-		case "PUT":
-			attrs.Verb = "update"
-		case "DELETE":
-			attrs.Verb = "delete"
-		default:
-			attrs.Verb = ""
-		}
-
-		vars := mux.Vars(r)
-		attrs.APIGroup = "core"
-		attrs.APIVersion = "v2"
-
-		// A non-default namespace is passed as a query parameter called "namespace"
-		namespace := r.URL.Query().Get("namespace")
-		if namespace == "" {
-			namespace = "default"
-		}
-		attrs.Namespace = namespace
+		attrs.ResourceName = vars["id"]
 
-		// Add the user to the attributes
-		if err := getUser(ctx, attrs); err != nil {
-			http.Error(w, err.Error(), http.StatusUnauthorized)
-			return
+		// Namespaces actually belong to the "default" namespace?
+		// Not forcing the namespace to "default" for namespaces leads to
+		// authorization failures.
+		// TODO: investigate
```

Comment:
Oh dear. I had thought that namespaces would belong to themselves. But when I reflect on what that means for the API, I'm not sure it makes any sense.